sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -410,13 +416,21 @@ static void appendCompileOptionsFromImage(std::string &CompileOpts,
     CompileOpts += isEsimdImage ? "-doubleGRF" : "-ze-opt-large-register-file";
   }
   // Add optimization flags.
-  // Add only if compile options are not overwritten by environment
-  // variable
-  if (!CompileOptsEnv && optLevel != -1) {
+  const char *optLevelStr = getUint32PropAsOptStr(Img, "optLevel");
+  // TODO: Passing these options to vector compiler causes build failure in
+  // backend. Will pass the flags once backend compilation issue is resolved.
+  if (!isEsimdImage && optLevelStr != nullptr && optLevelStr[0] != '\0') {
```

Comment:
Why did you drop `CompileOptsEnv` from the condition?

Reply:
For some reason, -cl-opt-disable gets added in the compilation pipeline to SYCL_PROGRAM_COMPILE_OPTIONS for -O0 compilation. I need to find out who is adding this and prevent it from being added as it is incorrect. I will follow up on this. Thanks for heads up.

Reply:
I have readded CompileOptsEnv. I was wrong to think compiler pipeline was setting this env var.

Thanks

Reply:
> For some reason, -cl-opt-disable gets added in the compilation pipeline to SYCL_PROGRAM_COMPILE_OPTIONS for -O0 compilation. I need to find out who is adding this and prevent it from being added as it is incorrect. I will follow up on this. Thanks for heads up.

@asudarsa, could you explain why it's incorrect, please?

Reply:
Hi @bader 

Sorry. That was a wrong analysis in the first place. -cl-opt-disable does not get added to SYCL_PROGRAM_COMPILE_OPTIONS by the compilation pipeline. That was a wrong analysis on my end. I have readded the check.

Thanks